jena-java/src/main/java/eu/neverblink/jelly/convert/jena/JenaEncoderConverter.java

```diff
@@ -2,10 +2,14 @@
 
 import eu.neverblink.jelly.core.NodeEncoder;
 import eu.neverblink.jelly.core.ProtoEncoderConverter;
+import eu.neverblink.jelly.core.utils.QuadDecoder;
+import eu.neverblink.jelly.core.utils.TripleDecoder;
 import org.apache.jena.graph.Node;
+import org.apache.jena.graph.Triple;
 import org.apache.jena.sparql.core.Quad;
 
-public final class JenaEncoderConverter implements ProtoEncoderConverter<Node> {
+public final class JenaEncoderConverter
+    implements ProtoEncoderConverter<Node>, TripleDecoder<Node, Triple>, QuadDecoder<Node, Quad> {
```

Comment:
So, this became a bit of an encoder/decoder naming mess.

ProtoEncoderConverter must stay as-is.

TripleDecoder and QuadDecoder aren't really doing any decoding, they are decomposing triples and quads. Decomposer? TripleTermExtractor & QuadTermExtractor?

Please also rename TripleEncoder and TripleDecoder while at it, it's not a good name for the same reason. TripleConstructor / QuadConstructor maybe? Or Factory?